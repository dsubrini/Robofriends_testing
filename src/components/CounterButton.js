import React, { PureComponent } from 'react';

class CounterButton extends PureComponent {
    constructor() {
        super();
        this.state = {
            count: 0
        }
    }

    updateCount = () => {
        this.setState(state => {
            return {count: state.count + 1}
        })
    };

    render() {
        return (
            <button color={this.props.color} onClick={this.updateCount} className="mb2 pa3">
                Count: {this.state.count}
            </button>
        )
    }
}

export default CounterButton;
