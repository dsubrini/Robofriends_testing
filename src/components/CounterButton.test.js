import { shallow } from 'enzyme';
import React from 'react';
import CounterButton from "./CounterButton";

it('expect to render CounterButton component', function () {
    const mockColor = 'red';
    const Component = shallow(<CounterButton color={mockColor}/>);
    expect(Component).toMatchSnapshot();

    const clikableButton = Component.find("button");
    clikableButton.simulate("click");
    clikableButton.simulate("click");
    expect(Component.state()).toEqual({count: 2});
    expect(Component.props().color).toEqual('red');
});
