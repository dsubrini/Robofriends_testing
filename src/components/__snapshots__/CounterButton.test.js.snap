// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CounterButton component 1`] = `
<button
  className="mb2 pa3"
  color="red"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
